### PR TITLE
code organization

### PR DESCRIPTION
**Pull request by ghost**
Code organization update

### CODE REVIEW

File: frontend/src/Apis/todos.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+export const urlString = "http://localhost:3001/todos/";
```
Comment (alinauman02) [inline frontend/src/Apis/todos.ts]: `urlString` does not need to be exported. The `App` component does not need to know the URL from where we fetch Todos. Remove `export` from this line, and update **ALL** functions below to not take a `url` parameter. They can use this variable for that, without needing an argument.

File: frontend/src/Apis/todos.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
```
Comment (alinauman02) [inline frontend/src/Apis/todos.ts]: All of these functions return something - the created/updated Todo, a list of Todos, a value telling that the Todo was deleted etc.

Modify these functions to `return` a `Promise` with the value being whatever the functions returns (e.g. A Todo, a list of Todos etc.).

In other words, the `App` component should not have to deal with the `Response` object, but instead it should directly get access to whatever the API returns.

**Review by alinauman02 — CHANGES_REQUESTED** [review]
Added a couple of comments for now.

Also there is an issue with files you moved:
- `src/components/` directory still has `Input.tsx`, `TodoList.tsx` and `TodoListItem.tsx` files. They should have been moved to their corresponding folders.
- The component files you have moved into directories have no git history.
- I said folders for React components use `PascalCase` naming convention. Other folders still use `kebab-case`. The folder you named `Apis` should have been named `apis`. Fix this.
- The `models` folder should also have a barrel file. Add one.

image

File: frontend/src/App.tsx [hunk 3]
```diff
@@ -1,12 +1,9 @@
 import "./App.css";
 import { FormEvent, useCallback, useEffect, useState } from "react";
```
Comment (ZunairaT) [inline frontend/src/App.tsx:1]: keep relative imports at the bottom and imports from ract other npm libraries at the top

**Review by ZunairaT — CHANGES_REQUESTED** [review]

File: frontend/src/App.tsx (hunk 3 above)
Comment (alinauman02) [reply frontend/src/App.tsx:1]: Just to expand on this, a relative import is one that begins with `.`, while an absolute import does not.

A practice we try to follow is to keep imports from npm modules (which are absolute imports) separated from those which are from our own code (which are usually relative, but we can make them absolute as well). Doing this makes it easy to understand which libraries a specific file/component depends on.

In this case, we would like to keep the line `import { FormEvent, useCallback, useEffect, useState } from "react";` at the top, followed by a blank line. Then we would add all imports from our own code, like `import "./App.css";`.

Check this for all the files.

**Review by ZunairaT — DISMISSED** [review]

**Review by alinauman02 — APPROVED** [review]

**Review by alinauman02 — CHANGES_REQUESTED** [review]
I just noticed a problem here. The second last and third last commits (SHA ecfa3e6229feab2491623a37e0321cde61017030 and 59988ba6b9e2165f55e3322f12b4762124454743) are not relevant to this PR/issue.

Either remove these from this PR or revert them so their changes do not come into this PR.

**Review by alinauman02 — APPROVED** [review]

**Pull request merged**